extractorapp/src/main/java/org/georchestra/extractorapp/ws/extractor/WfsExtractor.java

```diff
@@ -308,47 +326,52 @@ public void exceptionOccurred (Throwable exception) {
         return basedir;
     }
 
+	private SimpleFeatureCollection getFeatures(ExtractorLayerRequest request, SimpleFeatureType sourceSchema,
+			SimpleFeatureSource featureSource) throws IOException, TransformException, FactoryException {
+		
+		Query query = createQuery(request, sourceSchema);
+		SimpleFeatureCollection features = featureSource.getFeatures(query);
+		
+		CoordinateReferenceSystem returnedCrs = features.getSchema().getCoordinateReferenceSystem();
+		CoordinateReferenceSystem targetCrs = request._projection;
+		// current version (9.2) of WFS datastore does not perform reprojection
+		if (!CRS.equalsIgnoreMetadata(targetCrs, returnedCrs)) {
+			features = new ReprojectingFeatureCollection(features, targetCrs);
+		}
+		
+		return features;
+	}
+
 	/* This method is default for testing purposes */
     Query createQuery (ExtractorLayerRequest request, FeatureType schema) throws IOException, TransformException,
```

Comment:
I didn't find the "createQueryTest", may be private ?

Seems that there is no more IOException anymore.

Reply:
I didn't find it either, the method was already there. We should certainly add one though

Reply:
made it private and removed unnecessary IOException from throws list